krummstab/utils.py

```diff
@@ -75,6 +79,36 @@ def query_yes_no(text: str, default: bool = True) -> bool:
         return query_yes_no(text, default)
 
 
+# JSON parsing -------------------------------------------------------
+
+def validate_json(data: dict, schema: dict, source: str = "file",
+                  schema_version=jsonschema.Draft7Validator) -> None:
+    """
+    Validates a JSON object against a given schema.
+    """
+    try:
+        jsonschema.validate(data, schema, schema_version)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.critical(f"Validation error: {source} does not have "
+                         f"the right format: {error.message}")
+
+
+def read_json(source: pathlib.Path | Traversable | TextIO) -> dict:
```

Comment:
Make `read_json` either a string or a pathlib.Path. In the latter case, use `read_text` of pathlib to get string from file path.